doc/explanation/bootstrap.rst

```diff
@@ -0,0 +1,45 @@
+How Dune Uses Dune to Build Dune
+================================
+
+Dune's build system is itself Dune. This works thanks to a bootstrap process.
+This document explains how this works.
+
+Step 1: ``boot/bootstrap.ml``
+-----------------------------
+
+``boot/bootstrap.ml`` is an OCaml script (it is interpreted, not compiled) that
+is a mini-build system tailored to Dune itself. It computes dependencies
+between the various modules by calling ``ocamldep``, and it will generate build
+and link commands. It knows how to execute these commands in parallel. It does
+not read any ``dune`` file. However, the project structure and its system
+dependencies are encoded in ``boot/libs.ml``.
+
+This step produces ``_boot/dune.exe``.
+
+Step 2: Bootstrap Dune
+----------------------
+
+``_boot/dune.exe`` is the bootstrap Dune. Since it has been built from
+the Dune sources, it will act like Dune. However, since it has not been produced by
+Dune, it might not be reproducible or might contained unused modules (like
+platform-specific code).
+
+So the next step is to compile the Dune sources using the bootstrap Dune:
```

Comment:
Note that no compilation is does in this step. We just copy the bootstrapped dune into `_build/install`. This step is only used to generate the .install file.

Reply:
- wait, shouldn't we? this means that the installed dune has been produced by the bootstrap process, not by dune?
- Also, when we run `make dev`, we see that after the bootstrap, everything is re-built using an actual dune. is that just for the libs?

Reply:
> wait, shouldn't we? this means that the installed dune has been produced by the bootstrap process, not by dune?

We used to do that. It doubled the build times so users were relieved that we got rid of it.

> Also, when we run make dev, we see that after the bootstrap, everything is re-built using an actual dune. is that just for the libs?

For the libs, for merlin support, and of course for incremental builds.

Reply:
thanks a lot, I had missed that distinction. I'll incorporate that into the doc